### DISCUSSION TITLE
How does one remove a table from a datasource

### DISCUSSION THREAD

**Discussion by kpturner** [Budibase/budibase#10980] [Help] [answered]
I added a new datasource and then used 'fetch" to fetch all the tables.  I then realised it fetched a table that I did not want.  I cannot find an option to remove it?   If I fetch the tables again - this time specifically listing the tables I **do** want, it pulls those in but the previous erroneously fetched table is still there.  How does one remove it?

**Comment by ducnhoxlatui** [comment]
Hi @kpturner , 
You can try to delete all that datasource from budibase and then try to add again. When fetching, you need to check the "Fetch listed tables only (one per line)" and then typing in all tables that you want to fetch (one per line). 
![image](https://github.com/Budibase/budibase/assets/71976420/03b38186-ae91-4665-bd8a-a0273140cc32)

Hope this will help you ^^

**Comment by kpturner** [comment]
I deleted the datasource completely, added it back again, used "Fetch Tables" and listed only the ones I wanted.  The one I didn't want appeared again.   Is this info cached somewhere?

**Reply by ducnhoxlatui** [reply]
hmmm, that's weird, i never meet that situation TT

**Comment by kpturner** [comment]
I will try again.....

**Comment by kpturner** [comment]
So I tried again.  I deleted the datasource and logged out everywhere.
Logged back in and re-added the datasource.  All the previously fetched tables were still there (whereas normally it is an empty list).
This time I only requested one table:
![image](https://github.com/Budibase/budibase/assets/308646/69d1bf2f-fb82-45a4-b419-675cd645c71e)
Not only did it not remove the other tables (I would be surprised if it did), it gave me an error for a table I had not even listed
![image](https://github.com/Budibase/budibase/assets/308646/3ab83ade-5eba-4aa4-89f2-6f56fca57f44)

It makes sense to me that listing a specific table or tables does not remove other tables.  That would be bad.
It does not make sense to me that it apparently tried to fetch a table I had not even listed.
It does not make sense to me that, having deleted and re-added a datasource, that all the previous tables are still there.

I can only conclude that it is not possible to remove a table from the imported list once you have inadvertently added it?

**Comment by zedascouves** [comment]
I've encountered the same issue (and opened bug reports related to foreign keys). It seems there are major issues with relational databases and no one seems to be taking them into account.

**Comment by melohagan** [comment, marked as answer]
Hey @kpturner 

As of version **2.8.1** you can now select the tables that you wish to import. Any tables that are not selected will be removed from your table list if present.

**Reply by kpturner** [reply]
So if I want to remove a table I have to specifically re-import all the others except the one I want to delete?  What happens to any schema changes I have made to the existing good tables - like changing a column to an "options" column etc?

**Reply by melohagan** [reply]
The meta data, such as changing a type to options, should be kept even when you re-fetch the tables. Did a quick test there now to double check.
